DPL: improve Signposts implementation

ktf opened the pull request:
DPL: improve Signposts implementation

- make remove_engineering_type helper function consteval
- Remove engineering type when invoking the RAW function

alibuild commented:
Error while checking build/O2/fullCI for 149d36257f487c096c69852a98e4766d26325c6f at 2024-04-29 20:20:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 58>, long int&)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 58>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 58>, int64_t&)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 58>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 40>)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 40>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 61>, int)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 61>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 77>, int)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 77>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 85>)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 85>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 74>)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 74>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 91>)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 91>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 75>)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 75>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 28>)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 28>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 32>)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 32>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 51>)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 51>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 38>)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 38>, void>'
/sw/SOURCES/O2/13096-slc8_x86-64/0/Framework/Foundation/include/Framework/Signpost.h:118:38: error: no matching function for call to 'sprintf(std::array<char, 60>, std::vector<o2::framework::DataProcessingDevice::handleData(o2::framework::ServiceRegistryRef, o2::framework::InputChannelInfo&)::InputInfo>::size_type, fair::mq::Parts::size_type)'
/sw/slc8_x86-64/fmt/10.1.1-19/include/fmt/printf.h:603:11: error: no type named 'type' in 'struct fmt::v10::detail::char_t_impl<std::array<char, 60>, void>'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13096/149d36257f487c096c69852a98e4766d26325c6f/build_O2_fullCI/pretty.html).

ktf committed:
DPL: improve Signposts implementation

- make remove_engineering_type helper function consteval
- Remove engineering type when invoking the RAW function
